jetstream/core/orchestrator.py

```diff
@@ -242,64 +254,62 @@ def __init__(
     # that slot int should from now be placing tokens in the return channel of
     # the ActiveRequest.
     # If it receives (int, sample_tokens) then it actually
-    # does a detokenisation for any slots which have previously been set active
+    # does a detokenization for any slots which have previously been set active
     # via the previous kind of object, and the int is used to log which step
     # the tokens were created at. By having them in one queue we prevent
     # the possibility of race conditions where a slot is made live before the
     # tokens are ready and it receives tokens from a different sequence,
     # or tokens detokenized before the relevant slot is live.
-    self._detokenize_backlogs = [queue.Queue() for _ in generate_engines]
-    # Finally we have a queue that doesn't represent a stage, but tracks the
-    # internal status of a request within the generation loop.
-    # a) A queue of integers representing available 'slots' in the decode
-    #   operation. I.e. potentially available rows in the batch and/or
-    #   microbatch. When we want to insert a prefill result, we pop an integer
-    #   to insert at. When this is empty, it means all slots are full.
-
-    # Construct a)
-    self._generate_slots = [queue.Queue() for _ in generate_engines]
+    self._detokenize_backlogs = [
+        # We don't let detokenization accumulate more than 8 steps to avoid
+        # synchronization issues.
+        queue.Queue(8)
```

Comment:
How did you comeup with 8? Is that hardware dependent -- will this work for all configs (v5e-8, v5e-1 etc?)

Reply:
The JET has an investigation on GIL contention issue and detokenize queue size, and figured out that this size can avoid synchronization issue. It's not dependent to hardware.